### PR TITLE
Add body length check for Django's capture_body

### LINKED ISSUE #1478: APM agent crashes if the request body is too large
**Describe the bug**: ...
When the request body is too large, reporting to the apm server fails. 
The error message is: 
```
eapm event processor thread elasticapm.transport:286 - Failed to submit message: 'HTTP 400: {"accepted":4,"errors":[{"message":"event exceeded the permitted size." ...
```

**To Reproduce**

1. Check the body's size before reporting. If it's too large, report the body as `[Too large]` or something like that.

**Environment (please complete the following information)**
- OS: mac
- Python version: 3.7
- Framework and version [e.g. Django 2.1]: django==2.2.26
- APM Server version: ?
- Agent version: 5.10.1

Elastic apm configuration (in `settings.py`): 
```
ELASTIC_APM = {
	...
    "CAPTURE_BODY": "all",
}
```

**Additional context**

The issue is that Django client extracts the request body [without checking the size of the body](https://github.com/elastic/apm-agent-python/blob/main/elasticapm/contrib/django/client.py#L151-L152). I believe that adding a size check in [this](https://github.com/elastic/apm-agent-python/blob/97188c362497910762e5b1ef8fc53e4011ec75b8/elasticapm/contrib/django/client.py#L117) function should solve the issue.

### PR DESCRIPTION
**Pull request by basepi**
## What does this pull request do?

We had no restrictions on the body length in Django. This could result in the event payload being rejected by the APM Server.

As an aside, we'll need to figure out a better solution in the long term. (#305)

## Related issues
closes #1478

### CODE REVIEW

File: elasticapm/contrib/django/client.py [hunk 1]
```diff
@@ -149,7 +150,12 @@ def get_data_from_request(self, request, event_type):
                         self.logger.debug("Can't capture request body: %s", str(e))
                         data = "<unavailable>"
                 if data is not None:
-                    result["body"] = data
+                    # This can be removed once we figure out a soluction to
+                    # https://github.com/elastic/apm-agent-python/issues/305
+                    if len(str(data)) > constants.KEYWORD_MAX_LENGTH:
```
Comment (trentm) [inline elasticapm/contrib/django/client.py:155]: There is https://github.com/elastic/apm/blob/main/specs/agents/field-limits.md#long_field_max_length-configuration in the specs. Is that something that is better done as part of #305?

File: tests/contrib/django/django_tests.py [hunk 2]
```diff
@@ -848,7 +848,7 @@ def test_request_metrics_301_append_slash(django_elasticapm_client, client):
         **middleware_setting(
             django.VERSION,
             ["elasticapm.contrib.django.middleware.TracingMiddleware", "django.middleware.common.CommonMiddleware"],
-        )
+        ),
```
Comment (trentm) [inline tests/contrib/django/django_tests.py:851]: Are these changes from a code-formatter change from a while ago that is only applied as a particular file is updated?

File: elasticapm/contrib/django/client.py [hunk 3]
```diff
@@ -149,7 +150,12 @@ def get_data_from_request(self, request, event_type):
...
+                        result["body"] = keyword_field(str(data))
```
Comment (trentm) [inline elasticapm/contrib/django/client.py:156]: Is there a potential perf concern with `str(data)`ing twice?

**Review by trentm — APPROVED** [review]

File: elasticapm/contrib/django/client.py (hunk 1 above)
Comment (basepi) [reply elasticapm/contrib/django/client.py:155]: Ah, I had not realized that it was extended to more than `db.statement` -- I found the old `apm-dev` thread but didn't realize it had been made applicable to request body. I'll update and add that config.

File: tests/contrib/django/django_tests.py (hunk 2 above)
Comment (basepi) [reply tests/contrib/django/django_tests.py:851]: Correct. I touched the file so `black` hit it. We don't run `black` on the test suite, a decision made before I joined Elastic that could probably be revisited.

File: elasticapm/contrib/django/client.py (hunk 3 above)
Comment (basepi) [reply elasticapm/contrib/django/client.py:156]: Hmm. Possibly. I'll store it in a temporary variable.

**Review by trentm — APPROVED** [review]

**Pull request merged**